nncf/quantization/algorithms/min_max/openvino_backend.py

```diff
@@ -138,10 +138,17 @@ def create_convert_insertion_command(
     def get_target_point_shape(nncf_graph: NNCFGraph, node: NNCFNode, target_point: OVTargetPoint) -> Tuple[int, ...]:
         if target_point.is_weight_target_point():
             return node.layer_attributes.constant_attributes[target_point.port_id]["shape"]
+
+        # NOTE: Assumes that all output/input edges for the `node` with `output_port_id`/`input_port_id`
+        # equal to `target_point.port_id` should have the same `tensor_shape` value.
+
         if target_point.type == TargetType.PRE_LAYER_OPERATION:
-            return nncf_graph.get_input_edges(node)[target_point.port_id].tensor_shape
+            edges = [e for e in nncf_graph.get_input_edges(node) if e.input_port_id == target_point.port_id]
+            return edges[0].tensor_shape
         elif target_point.type == TargetType.POST_LAYER_OPERATION:
-            return nncf_graph.get_output_edges(node)[target_point.port_id].tensor_shape
+            edges = [e for e in nncf_graph.get_output_edges(node) if e.output_port_id == target_point.port_id]
+            return edges[0].tensor_shape
+
```

Comment:
The following statement

```python
nncf_graph.get_output_edges(node)[target_point.port_id]
```
it is an incorrect usage of the API because the index in the `nncf_graph.get_output_edges(node)` list is not the same as `target_point.port_id`. For example, the `nncf_graph.get_output_edges(node)` list may contain only one edge that connects `output_port_id = 1` of a `node` with some `input_port_id` of node `U` (LSTMSequence case). In that case, `target_point.port_id = 1`, but the list contains only one element with index 0.